src/test/java/io/arlas/server/rest/tag/TagIT.java

```diff
@@ -0,0 +1,109 @@
+ */
+
+package io.arlas.server.rest.tag;
+
+import io.arlas.server.AbstractTestWithCollection;
+import io.arlas.server.CollectionTool;
+import io.arlas.server.model.request.Tag;
+import io.arlas.server.model.request.TagRequest;
+import io.arlas.server.rest.explore.SearchServiceIT;
+import org.junit.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.everyItem;
+
+public class TagIT extends AbstractTestWithCollection {
+
+    @Override
+    public String getUrlPath(String collection) {
+        return arlasPath + "write/"+collection+"/_tag";
+    }
+
+    @Test
+    public void testAddTag() throws InterruptedException {
+        TagRequest tr = new TagRequest();
+        tr.tag=new Tag();
+        tr.tag.path="params.tags";
+        tr.tag.value="v1";
+
+        given().contentType("application/json")
+                .body(tr)
+                .when()
+                .post(getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200);
+
+        Thread.currentThread().sleep(5000);
+
+        given()
+                .get(new SearchServiceIT().getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200)
+                .body("hits[0].data.params.tags.size()",equalTo(1))
+                .body("hits.data.params.tags[0]", everyItem(equalTo("v1")))
+        ;
+
+        tr.tag.value="v2";
+        given().contentType("application/json")
+                .body(tr)
+                .when()
+                .post(getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200);
+
+        Thread.currentThread().sleep(5000);
+
+        given()
+                .get(new SearchServiceIT().getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200)
+                .body("hits[0].data.params.tags.size()",equalTo(2))
+                .body("hits[0].data.params.tags[0]", (equalTo("v1")))
+                .body("hits[0].data.params.tags[1]", (equalTo("v2")))
+        ;
+    }
+
+
+    @Test
+    public void testAddTagOnSingleValue() throws InterruptedException {
+        TagRequest tr = new TagRequest();
+        tr.tag=new Tag();
+        tr.tag.path="params.job";
+        tr.tag.value="Another job";
+
+        given().contentType("application/json")
+                .body(tr)
+                .when()
+                .post(getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200);
+
+        Thread.currentThread().sleep(5000);
+
+        given()
+                .get(new SearchServiceIT().getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200)
+                .body("hits[0].data.params.job.size()",equalTo(2))
+                .body("hits[0].data.params.job[1]", equalTo("Another job"))
+        ;
+    }
```

Comment:
Add a test function with invalid tags paths/values to make sure that exceptions will be thrown

Reply:
ok